redis-desktop-manager: Fix 'missing plugin "xcb"'

roblabla opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

The current redis desktop manager fails to start when installed with `nix-env`. The error is:

```
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
```

This appears to be due to using stdenv.mkDerivation instead of the mkDerivation set by the `libsForQt5.callPackage`. Furthermore, after fixing this, another issue came up: The application would fail to start due to a missing dependency on qtquickcontrols2. This was fixed by adding the missing dependency.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

roblabla committed:
redis-desktop-manager: Fix 'missing plugin "xcb"'

The current redis-desktop-manager derivation fails to start with the
following error:

  Could not find the Qt platform plugin "xcb" in ""

The error seems to be that stdenv.mkDerivation is used instead of
whatever override is used by qt5.callPackage.

roblabla committed:
redis-desktop-manager: Add missing qtquikctrl2 dep

Adds a missing dependency on qtquickcontrols2. Fixes runtime failure.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 113293` at c062f1c8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/32a35062b05ca3cc15de3ab9ffa266da">redis-desktop-manager</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/misc/redis-desktop-manager/default.nix:43:3:
```
   |
43 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/misc/redis-desktop-manager/default.nix:70:3:
```
   |
70 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl21` is a deprecated license, check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/applications/misc/redis-desktop-manager/default.nix:79:5:
```
   |
79 |     license = licenses.lgpl21;
   |     ^
```
</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/applications/misc/redis-desktop-manager/default.nix:
```diff
@@ -28,7 +28,7 @@ stdenv.mkDerivation rec {
   buildInputs = [
     libssh2 qtbase qtdeclarative qtgraphicaleffects qtimageformats
-    qtquick1 qtquickcontrols qtsvg qttools qtcharts
+    qtquick1 qtquickcontrols qtquickcontrols2 qtsvg qttools qtcharts
```
do we need qtquickcontrols and qtquickcontrols2?

roblabla replied:
I'm pretty sure it does. it uses controls from both versions ([ImageButton](https://github.com/uglide/RedisDesktopManager/blob/0.9.1/src/qml/common/ImageButton.qml#L2) uses v2, [most other controls](https://github.com/uglide/RedisDesktopManager/blob/0.9.1/src/qml/common/AddressInput.qml#L3) uses v1). To double check, I'm trying to build without the `qtquickcontrols`, but I expect it to fail to work.

roblabla replied:
Hmm, nvm , you appear to be right, just using qtquickcontrols2 appear to be enough.

roblabla committed:
redis-desktop-manager: remove unnecessary dep

The qtquickcontrols dependency appears to be unneeded, as it is
redundant with the qtquickcontrols2 dependency.

roblabla committed:
redis-desktop-manager: Fix license.

RedisDesktopManager uses GPLv3 since version 0.9.0.
